hbase-client/src/main/java/org/apache/hadoop/hbase/client/ZKConnectionRegistryURIFactory.java

```diff
@@ -30,9 +34,34 @@
  * Connection registry creator implementation for creating {@link ZKConnectionRegistry}.
  */
 @InterfaceAudience.Private
-public class ZKConnectionRegistryCreator implements ConnectionRegistryURIFactory {
+public class ZKConnectionRegistryURIFactory implements ConnectionRegistryURIFactory {
 
-  private static final Logger LOG = LoggerFactory.getLogger(ZKConnectionRegistryCreator.class);
+  private static final Logger LOG = LoggerFactory.getLogger(ZKConnectionRegistryURIFactory.class);
+
+  public static final String SESSION_TIMEOUT = "session.timeout.ms";
```

Comment:
While I appreciate the convenience of these shortened configuration strings, I think that we should also accept the full configuration key as it is read from hbase-site.xml.

Reply:
This is just another choice comparing to #5853 . Since you and @bbeaudreault both think allowing users to override all configuration through URI queries is acceptable, I think we could go #5853 .